src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -43,6 +48,9 @@ private static string MarshalToString(IntPtr arg, string argName)
                                                                    IntPtr reserved,
                                                                    IntPtr functionHandle)
         {
+            if (!IsSupported)
+                return -1;
```

Comment:
This returns HRESULT - so we should return at least E_FAIL - but ideally something better. Maybe add a new one to https://github.com/dotnet/runtime/blob/main/docs/design/features/host-error-codes.md (there's an enum in the native host for this) and then use it here. Since this error is directly propagated to the native caller, it makes sense to have it return native hosting error code. The best solution would be to have a distinct failure - so that it's easier to debug when it happens.

Reply:
I like an existing value that is already there, HostApiUnsupportedScenario, since that seems appropriate just going by the name. Let me know if we should define a new one for this

Reply:
I personally would like to introduce new one - something like HostFeatureDisabled or so... to denote that it is something which can be changed (unlike unsupported - which is always unsupported).